docs/FRSKY_R9.md

```diff
@@ -70,18 +84,84 @@ The wiring is similar to the ExpressLRS docs found in the link below. Different
+
+3. Connect a TTL USB serial adapter to your computer and install a driver if it's not recognized.  If you don't have a TTL USB serial adapter, you should be able to use your flight controller.  See step 8
+
+4. Wire the serial adapter to your receiver as instructed [here](https://www.expresslrs.org/quick-start/receivers/r9/#wiring-up-your-receiver).  But, don't connect the VCC wire yet.
+
+5. Use the command window to run the UARTupload.py script like: "python UARTupload.py image\_file\_path" where image\_file\_path is the path to the appropriate .elrs file you obtained in step 1.
+
+6. When the python script reports "attempting to reboot into bootloader", power up the receiver by connecting the VCC wire to the 5 volt output of your serial adapter or a 5 volt power supply.  You should see the script report sync and begin the firmware download.  If it fails, try again, the timing can be a bit tight.  The first time you flash the .elrs file, you may not need to delay connecting the VCC wire.  If you have trouble getting the timing correct, you can hold down the button when powering up the receiver and the ELRC bootloader will keep running and not start the application code.
+
+7. When the UARTupload.py script reports the flash was successful, you can leave the receiver powered by the serial adapter and try to establish a connection from your TX.  The LED will switch from rapid red to 1Hz green on both the RX and TX when the connection is established.
+
+8. After you have connected the receiver serial port to your flight controller and installed it in your build, you can use Ardupilot's [serial passthrough](https://ardupilot.org/copter/docs/common-serial-passthrough.html) feature for future mLRS firmware updates without uninstalling the receiver from your build.  The baud rate for the FC serial port used should be configured for 420,000 in Ardupilot.  The receiver should be powered up (battery connected) after the UARTupload.py script reports "attempting to reboot into bootloader" as in step 6.  This is most easily accomplished by wiring the receiver to a regulated power output on your flight controller or ESC which is active only when the battery is connected, but not powered when the flight controller is connected to USB without the battery.  If this can't easilly be arranged, you will have to hold down the receiver button when connecting the flight controller USB port.
+
+### Flashing via ST-Link ###
 
-Note: Fashing mLRS with the ST-Link is a non-reversible operation, i.e., it is not possible to revert back to the original Frsky firmware. It is possible to switch to ExpressLRS however.
+Note: Fashing any Frsky R9 board with the ST-Link is a non-reversible operation, i.e., it is not possible to revert back to the original Frsky firmware. It is possible to switch to ExpressLRS however.
```

Comment:
Spelling - 'Flashing'

Reply:
it seems to me this will need 2 or 3 iterations, there are a number of pure language thingies :)
but we will get it done. It's really amazing work by Brad.

Reply:
maybe we should have a video chat to go through the text line by line. Might be faster&quicker than repeated comments.

Reply:
I think this entire section could stand a re-write as it seems a bit confusing.  I also think it might discourage people a bit too much from trying the cheap dongles.  I have not had nearly the trouble with the cheap ST-Link dongles as Olli has had.  I have owned a couple of different brands and they both worked reliably as long as I keep the wires very short.  The only trouble I've had was related to neglecting to update their firmware when it was required.  I also damaged the first one I owned by neglecting to unplug it before powering on the device I had programmed.

I think it's still worth mentioning there could be trouble, but I'm not sure how to state this.

Reply:
I think with the ELRS bootloader as the default go-to method the STLink is clearly just for the "experts" ... and we can leave it top them or expect them to sort ourt whatever issues they may have ... i.e., pl feel free to cut it down and strip it off blunder ... i.e., I wouldn't even mention there could be problems ... just refer to STLink programmer, and done

Reply:
adding to this, I think you could delete everything from "However, there might be catches: ..." onwards.

Maybe one could finsih the sentence before with "... and the new recommended tool is "STM32CubeProgrammer, which we also recommend to use".